src/serac/physics/materials/thermal_material_input.cpp

```diff
@@ -0,0 +1,106 @@
+      if (density_present && cp_present && !kappa_present && dim_present && kappa_tensor_present) {
+        // Verify rows of kappa tensor struct is an array of doubles and is of proper size
+        int  dim          = c["dim"];
+        bool row1_present = c.contains("kappa_tensor/row1") && (c["kappa_tensor/row1"].type() == coll_type);
+        bool row2_present = c.contains("kappa_tensor/row2") && (c["kappa_tensor/row2"].type() == coll_type);
+        bool row3_present = c.contains("kappa_tensor/row3") && (c["kappa_tensor/row3"].type() == coll_type);
+        auto row1_size    = c["kappa_tensor/row1"].get<std::vector<double>>().size();
+        auto row2_size    = c["kappa_tensor/row2"].get<std::vector<double>>().size();
+        auto row3_size    = c["kappa_tensor/row3"].get<std::vector<double>>().size();
+
+        if (dim == 2) {
+          return row1_present && (row1_size == 2) && row2_present && (row2_size == 2) && !row3_present;
+        } else if (dim == 3) {
+          return row1_present && (row1_size == 3) && row2_present && (row2_size == 3) && row3_present &&
+                 (row3_size == 3);
+        }
+      }
+    }
+
+    return false;
+  });
+}
+
+}  // namespace serac
+
+std::vector<std::vector<double>> FromInlet<std::vector<std::vector<double>>>::operator()(
+    const axom::inlet::Container& base)
+{
+  std::vector<std::vector<double>> result;
+
+  result.push_back(base["row1"].get<std::vector<double>>());
+  result.push_back(base["row2"].get<std::vector<double>>());
+  result.push_back(base["row3"].get<std::vector<double>>());
+
+  return result;
+}
+
+serac::var_thermal_material_t FromInlet<serac::var_thermal_material_t>::operator()(const axom::inlet::Container& base)
+{
+  serac::var_thermal_material_t result;
+  std::string                   model = base["model"];
+
+  if (model == "LinearIsotropicConductor") {
+    result = serac::heat_transfer::LinearIsotropicConductor(base["density"], base["cp"], base["kappa"]);
+  } else if (model == "LinearConductor") {
+    // Store tensor in a vector temporarily, then
+    // set the tensor values and material result based on the dimension
+    int dim = base["dim"];
+    if (dim == 2) {
+      std::vector<std::vector<double>> v    = {base["kappa_tensor"]["row1"], base["kappa_tensor"]["row2"]};
+      serac::tensor<double, 2, 2>      cond = {{{v[0][0], v[0][1]}, {v[1][0], v[1][1]}}};
+      result = serac::heat_transfer::LinearConductor<2>(base["density"], base["cp"], cond);
+    } else if (dim == 3) {
+      std::vector<std::vector<double>> v    = {base["kappa_tensor"]["row1"], base["kappa_tensor"]["row2"],
+                                            base["kappa_tensor"]["row3"]};
+      serac::tensor<double, 3, 3>      cond = {
+          {{v[0][0], v[0][1], v[0][2]}, {v[1][0], v[1][1], v[1][2]}, {v[2][0], v[2][1], v[2][2]}}};
+      result = serac::heat_transfer::LinearConductor<3>(base["density"], base["cp"], cond);
+    }
+  }
```

Comment:
is it possible to query the type associated with a lua variable? If so, we could implement something like

```cpp
if(is_a_double(base["kappa"])) {
  result = serac::heat_transfer::LinearIsotropicConductor(base["density"], base["cp"], base["kappa"]);
}
if(is_an_array_of_arrays(base["kappa"])) {
  int num_rows = base["kappa"].size();
  if (dim == 2) {
      std::vector<std::vector<double>> v    = {base["kappa_tensor"]["row1"], base["kappa_tensor"]["row2"]};
      serac::tensor<double, 2, 2> conductivity = {{{v[0][0], v[0][1]}, {v[1][0], v[1][1]}}};
      result = serac::heat_transfer::LinearConductor<2>(base["density"], base["cp"], conductivity);
  }
  if (dim == 3) {
      std::vector<std::vector<double>> v    = {base["kappa"]["row1"], base["kappa"]["row2"], base["kappa"]["row3"]};
      serac::tensor<double, 3, 3> conductivity = {{
        {v[0][0], v[0][1], v[0][2]}, 
        {v[1][0], v[1][1], v[1][2]}, 
        {v[2][0], v[2][1], v[2][2]}
      }};
      result = serac::heat_transfer::LinearConductor<3>(base["density"], base["cp"], conductivity);
  }
}
```

Then the user wouldn't need to explicitly name the kind of LinearConductor it is in the input file, e.g.

```lua
materials = {
  -- we can tell that it's an isotropic model (LinearIsotropicConductor) since kappa is a floating point number
  { model = "LinearConductor", kappa = 0.5, cp = 1.0, density = 1.0 }, 

  -- we use the more general material model (LinearConductor) when kappa is a matrix
  -- also, we can figure out dim = 2 from the matrix dimensions, rather than asking the user to put that in the input file
  { model = "LinearConductor", kappa = [[1.0, 0.0], [0.0, 2.0]], cp = 1.0, density = 1.0 } 
}
```

Reply:
We could also do something similar with the J2 models, so that the users just write "J2" in the model name, and then look at the "hardening" variable to figure out which one to use (or just remove our old "J2" model and replace it by a linear hardening rule in the more general model).

Reply:
The problem is that we need to define `kappa`'s type in `ThermalMaterialInputOptions::defineInputFileSchema()` before we read from a lua file, which is why I unfortunately had to use different names.

Reply:
> The problem is that we need to define kappa's type in ThermalMaterialInputOptions::defineInputFileSchema() before we read from a lua file

Where does this requirement come from? If I comment out 
```
container.addDouble("kappa", "The conductivity parameter");
```
am I not allowed to have a field `kappa` for that material model in the input file?

Reply:
Yes because then you wouldn't be able to access it in the `FromInlet operator()` like `base["kappa"]`.

Reply:
This is a good idea and I believe it will require modifications in Inlet.

Reply:
issue in axom tracking this https://github.com/LLNL/axom/issues/1234